File: LeetCode/py/1309.py
#Link: https://leetcode.com/problems/decrypt-string-from-alphabet-to-integer-mapping/

class Solution:
    def freqAlphabets(self, s: str) -> str:
        i = 0
        result = ''
        while i < len(s):
            if i < len(s)-2 and s[i+2] == '#':
                value = int(s[i:i+2])
                i += 2
            else: value = int(s[i])
            result += chr(97 + value - 1)
            i += 1
        return result
        

File: LeetCode/cpp/1309.cpp
//Link: https://leetcode.com/problems/decrypt-string-from-alphabet-to-integer-mapping/

class Solution {
public:
    string freqAlphabets(string s) {
        string result;
        int value;
        for(int i = 0 ; i < s.length() ; i++) {
            if(i < s.length()-2 && s[i+2] == '#') {
                value = (s[i]-48)*10 + (s[i+1]-48);
                i += 2;
            }
            else value = s[i]-48;
            result += (97 + value - 1);
        }
        return result;
    }
};
